### PR TITLE
fix: align Encoding APIs to spec using WPT

### PR DESCRIPTION
**Pull request by lucacasonato**
Part of #9001

### CODE REVIEW

File: cli/tests/integration_tests.rs [hunk 1]
```diff
@@ -4963,6 +4967,35 @@ fn concat_bundle(files: Vec<(PathBuf, String)>, bundle_path: &Path) -> String {
   bundle
 }
 
+// TODO(lucacasonato): DRY with tsc_config.rs
+/// Convert a jsonc libraries `JsonValue` to a serde `Value`.
```
Comment (bartlomieju) [inline cli/tests/integration_tests.rs:4971]: We should implement `TryFrom` in `jsonc` library

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/tests/integration_tests.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/tests/integration_tests.rs:4971]: Opened https://github.com/dprint/jsonc-parser/issues/5

File: cli/tests/unit/text_encoding_test.ts [hunk 2]
```diff
@@ -69,31 +69,7 @@ unitTest(function textDecoder2(): void {
   assertEquals(decoder.decode(fixture), "𝓽𝓮𝔁𝓽");
 });
```
[the remark this answers is not in the document]
Comment (lucacasonato) [reply cli/tests/unit/text_encoding_test.ts:71]: I don't think so. WPT is way more exhaustive. We can remove these tests in a follow up.

File: op_crates/web/text_encoding_test.js [hunk 3]
```diff
@@ -1252,8 +1200,6 @@ function main() {
   atobThrows2();
   btoaFailed();
   textDecoder2();
```
[the remark this answers is not in the document]
Comment (lucacasonato) [reply op_crates/web/text_encoding_test.js:1202]: See above.

**Pull request merged**